src/authorization/WebAclReader.ts

```diff
@@ -147,122 +129,115 @@ export class WebAclReader extends PermissionReader {
   }
 
   /**
-   * Sets the correct values for non-acl permissions such as create and delete.
-   * Also adds the correct values to indicate that having control permission
-   * implies having read/write/etc. on the acl resource.
+   * Finds the ACL data relevant for all the given resources.
+   * The input set will be modified in place.
+   *
+   * @param targets - Targets to find ACL data for.
    *
-   * The main reason for keeping the control value is so we can correctly set the WAC-Allow header later.
+   * @returns A map linking ACL resources to the relevant identifiers.
    */
-  private updateAclPermissions(aclPermissions: AclPermission, isAcl: boolean): AclPermission {
-    if (isAcl) {
-      return {
-        read: aclPermissions.control,
-        append: aclPermissions.control,
-        write: aclPermissions.control,
-        create: aclPermissions.control,
-        delete: aclPermissions.control,
-        control: aclPermissions.control,
-      };
+  private async getAclMatches(targets: Set<ResourceIdentifier>): Promise<Map<AclMatch, ResourceIdentifier[]>> {
```

Comment:
```suggestion
  private async retrieveAclDocuments(targets: Set<ResourceIdentifier>): Promise<Map<AclMatch, ResourceIdentifier[]>> {
```

Reply:
But in general, I'm confused as to why we also return the stores here?
Couldn't `getAclRecursive` just get us the identifiers, which we read later then?

Maybe it used to be an optimization that we didn't need to perform a separate existence and read operation.